CMakeLists.txt

```diff
@@ -252,18 +251,18 @@ if (DEBUG)
       message(WARNING "No libBFD and/or libDL has been found!")
     endif()
   endif()
-else()
+elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
   add_definitions(-DNDEBUG)
   if (NOT MSVC)
     if (NOT CMAKE_CROSSCOMPILING)
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
-      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -O3")
+      set(CMAKE_CXX_FLAGS_RELEASE_INIT "${CMAKE_CXX_FLAGS} -O3")
+      set(CMAKE_C_FLAGS_RELEASE_INIT "${CMAKE_C_FLAGS} -std=c99 -O3")
     else()
-      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+      set(CMAKE_C_FLAGS_RELEASE_INIT "${CMAKE_C_FLAGS} -std=c99")
     endif()
   else ()
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /O3")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /O3")
+    set(CMAKE_CXX_FLAGS_RELEASE_INIT "${CMAKE_CXX_FLAGS} /O3")
+    set(CMAKE_C_FLAGS_RELEASE_INIT "${CMAKE_C_FLAGS} /O3")
   endif ()
 endif()
```

Comment:
According to the reading I've been doing, when `CMAKE_BUILD_TYPE` is `Release`, then the `NDEBUG` macro is automatically used, and `-O3` is specified, so actually we can just remove this whole block.